src/lib/components/ui/select/select.svelte

```diff
@@ -87,6 +75,38 @@
 			store.marked = store.value
 		}
 	})
+	let lastTabInteraction = $state(false)
+	let focused = $state(false)
+	$effect(() => {
+		const keyDown = (e: KeyboardEvent) => {
```

Comment:
Looks like this can live in the effect itself and actually despite my previous comment it may not need to be a state variable but just normal variable
```suggestion
	let focused = $state(false)
	$effect(() => {
		let lastTabInteraction = $state(false)
		const keyDown = (e: KeyboardEvent) => {
```